reactive/kubernetes_master.py

```diff
@@ -168,6 +168,35 @@ def channel_changed():
     set_upgrade_needed()
 
 
+# Returns True if a is subnet of b
+# This method is copied from cpython as it is available only from
+# python 3.7
+# https://github.com/python/cpython/blob/3.7/Lib/ipaddress.py#L1000
+def is_subnet_of(a, b):
+    try:
+        # Always false if one is v4 and the other is v6.
+        if a._version != b._version:
+            raise TypeError("{} and {} are not of the same version".format(
+                a, b))
+        return (b.network_address <= a.network_address and
+                b.broadcast_address >= a.broadcast_address)
+    except AttributeError:
+        raise TypeError("Unable to test subnet containment "
+                        "between {} and {}".format(a, b))
+
+
+def is_service_cidr_expansion():
+    service_cidr_from_db = db.get('kubernetes-master.service-cidr')
+    if service_cidr_from_db:
+        current_service_cidr = ipaddress.ip_network(service_cidr_from_db)
+        new_service_cidr = ipaddress.ip_network(hookenv.config('service-cidr'))
+        if not is_subnet_of(current_service_cidr, new_service_cidr):
```

Comment:
```suggestion
        if new_service_cidr == current_service_cidr:
            return False
        if not is_subnet_of(current_service_cidr, new_service_cidr):
```

Reply:
This change helps, but `is_service_cidr_expansion()` would still return True when `kubernetes-master.service-cidr` hasn't been set yet. I don't think that's what we want - we don't want to be deleting the kubernetes service when the cluster is first deployed.

Reply:
Good point.  We actually want different behaviors for `freeze_service_cidr()` vs the rolling restart, since the former should be done on the initial deploy but the latter shouldn't.

Reply:
Modified as per your suggestion